### ISSUE TITLE
3rd person view feedback!

### ISSUE THREAD

**Issue by ProPeach** [tsunamayo/Starship-EVO#4999]
Introduced in 22w45a experimental is a new 3rd person view for ships! This view is a little higher up and further back than before so you can more easily see what's in front of you. This post is for all feedback and suggestions on the new view.

For this plane, I think it works great.
![3rd person camera change](https://user-images.githubusercontent.com/27857711/201492776-c0401fdb-ec36-4114-8e9b-ef68712b948a.png)
When I made #4553 I only really tested the view on smaller ships, under ~30m and all quite flat. Now that the change is live, I think there could be some small tweaks done to improve the camera position on larger ships, and ships of different proportions.

For example - 
(All of these pictures were taken when the ship is at rest)

~75m long
https://steamcommunity.com/sharedfiles/filedetails/?id=2198932313
![711980_20221112201821_1](https://user-images.githubusercontent.com/27857711/201493013-6ecb86ee-de51-4ea6-bfc7-f55ad726392a.png)
https://steamcommunity.com/sharedfiles/filedetails/?id=2198932313

150m long
https://steamcommunity.com/sharedfiles/filedetails/?id=2386576566
![711980_20221112202154_1](https://user-images.githubusercontent.com/27857711/201493105-6ff5685c-0d3e-417d-a15f-a97ba2ebac84.png)

450m long
https://steamcommunity.com/sharedfiles/filedetails/?id=2633962530
![711980_20221112202321_1](https://user-images.githubusercontent.com/27857711/201493148-4d1e5484-76fa-41a6-9752-b4de7e0e7540.png)

1.2km long
https://steamcommunity.com/sharedfiles/filedetails/?id=2786070115
![711980_20221112202535_1](https://user-images.githubusercontent.com/27857711/201493221-d4d74c1f-5af1-4db6-9c66-53921f787862.png)

Ships that are tall are also a little bit too close, or too low

https://steamcommunity.com/sharedfiles/filedetails/?id=2786053583
![711980_20221112201504_1](https://user-images.githubusercontent.com/27857711/201493246-c635e4fc-8877-4b14-868c-ac889670f338.png)

![711980_20221112202920_1](https://user-images.githubusercontent.com/27857711/201493407-2d817b00-5ab2-40a4-bced-a5b0d7004d84.png)
[Stinger Mantis 2.zip](https://github.com/tsunamayo/Starship-EVO/files/9995871/Stinger.Mantis.2.zip)

_**Dynamic Camera**_
Separate from the camera position is the way it moves when you fly the ship, but both of these work together to give the flying experience. I think this dynamic movement could also use some tweaks.

https://i.imgur.com/CIvKxmb.mp4
The TIE from earlier
While the resting position for the camera was a bit too close, the position while the ship is flying forward feels nice. But I think the max range that the camera can move backwards and forwards due to ship velocity is too much, perhaps reduce the upper and lower bounds.

https://i.imgur.com/CuhpuVR.mp4
The same 450m ship from above, doing some manoeuvring with the mouse and WASD. The camera distance moves about is huge, almost the whole length of the ship, and it moves very quickly. But the ship does not move much at all. Is it possible to scale the amount the dynamic camera moves, with the agility of the ship? I think that'd improve the feel of piloting different size ships much more interesting and varied.

Looking forward to seeing examples from other players and their ships too with the new new camera!

**Comment by ProPeach** [comment]
Here are some examples of camera positions on large ships in other games -
Fractured Space 
![image](https://user-images.githubusercontent.com/27857711/201494137-53363bb2-b790-4ff0-9d9d-b8733b5b91d4.png)
https://www.gameogre.com/wp-content/uploads/2016/05/fracturedspace2.jpg

**Comment by Dwarf-LordPangolin** [comment]
Thanks for setting this up Peach! And thank you Tsuna for moving the camera; it's definitely an improvement over where it was before.

I agree with Peach that it's a little too far now. This is how it looks with a ship I'm currently working on:
![20221112155857_1](https://user-images.githubusercontent.com/33041786/201496273-c6fa60b2-39fe-4834-8ee0-561b541d7680.jpg)

Would it be possible to create a couple configuration options for the camera's height and distance? Either in the Graphics Options, or even better yet on a per-ship basis?

**Comment by kynamats** [comment]
I agree that it's definitely an improvement on before, its really noticeable in combat. I think however that there will always be too much variation in geometry, ship type and personal preference, that I feel the only solution is a slider for the camera angle in the starship computer interface - this would allow people to set up camera angles for their created and shared ships, and for other people to tweak them to their preference in their persistent ships, when survival comes.

**Comment by DrizztCode** [comment]
only played a little bit with the new build - but i definitely prefer the camera further away like this - it was definitely too close before - whereas now it is actually decent to fly with

but i do agree that it may be a little far - but that it will always be subjective - both for each player and depending on the ship size and shape

i do think that some customisability for it would be cool - but i imagine that the zoom affect used when accelerating may complicate that somewhat

but either way - definitely is a big improvement! great work and many thanks! :-)

**Comment by BigBadKangaroo** [comment]
I've noticed aswell that the camera is fine on smaller vessels, but really gets a bit too low on larger ones. Moving CE's make a difference for the position of the camera as well:

![20221115230125_1](https://user-images.githubusercontent.com/81495956/202034676-57ccaac8-fa05-4cf6-8930-e206ea8e1769.jpg)
![20221115230115_1](https://user-images.githubusercontent.com/81495956/202034775-77222623-ffe1-4f06-843d-ea6a5f485230.jpg)
![20221115230152_1](https://user-images.githubusercontent.com/81495956/202034791-0a9ce62a-2b7c-4d72-972f-e7027513bee6.jpg)

**Comment by Servomotor** [comment]
I am in opposition to most of the posts. I hate the new angle. I found the original camera angle just fine. 

The issue I am having now, is if you fly a ship into a hangar, drydock etc, the camera has the strong potential to be above the hangar/drydock, which makes it hard to get out of. 

This is an unpopular opinion, but frankly I think all flight should take place in first person for balance. 

Anyway. This is my ship in front of the drydock, then under the drydock (where you see clipping) and then a side view showing the size difference if that helps. 

![20221116220856_1](https://user-images.githubusercontent.com/65745916/202353761-201e6763-6d0b-427a-9b42-169cf3d564f7.jpg)
![20221116220914_1](https://user-images.githubusercontent.com/65745916/202353770-e9b2e995-ee3a-4639-a7fc-96df1eac9beb.jpg)
![20221116220921_1](https://user-images.githubusercontent.com/65745916/202353780-d1491e9c-9a6c-4aa3-8e2e-44c5d147926f.jpg)
![20221116220950_1](https://user-images.githubusercontent.com/65745916/202353786-9aa5372d-560f-450a-a0b6-c3bc6cb15fa6.jpg)

**Comment by Uncle-Ulty** [comment]
Guys, third person camera is something FFXIV 2.0 solved back in 2013

HOW? press ctrl + up/down, to adjust camera height

simple as that

this way each player can config it's own camera

**Comment by Servomotor** [comment]
Ya, it should be something that we can do on the fly (no pun intended...or was it?!?!?)

**Comment by Dwarf-LordPangolin** [comment]
I don't know if that would be easy for Tsuna, but I agree: that would indeed be the best of both worlds.

Regarding balance concerns: Space Engineers actually has an option to disable the third-person camera in its settings when creating a new world. If that is indeed a balance concern, perhaps SEVO could have that option as well?

**Comment by Servomotor** [comment]
> I don't know if that would be easy for Tsuna, but I agree: that would indeed be the best of both worlds.
> 
> Regarding balance concerns: Space Engineers actually has an option to disable the third-person camera in its settings when creating a new world. If that is indeed a balance concern, perhaps SEVO could have that option as well?

That is a good idea for world generation/ruleset.

**Comment by JRL101** [comment]
> I've noticed aswell that the camera is fine on smaller vessels, but really gets a bit too low on larger ones. Moving CE's make a difference for the position of the camera as well:
> 
> ![20221115230125_1](https://user-images.githubusercontent.com/81495956/202034676-57ccaac8-fa05-4cf6-8930-e206ea8e1769.jpg) ![20221115230115_1](https://user-images.githubusercontent.com/81495956/202034775-77222623-ffe1-4f06-843d-ea6a5f485230.jpg) ![20221115230152_1](https://user-images.githubusercontent.com/81495956/202034791-0a9ce62a-2b7c-4d72-972f-e7027513bee6.jpg)

I do like the ships rear being visible in my space games, most times i bring the 3rd person ship view to the likeness of a First person shooter, where the ship is your weapon, but having it off to the side like a weapon feels uncomfortable, and having the thrusters completely obscured just removes your ship from everything visually. Having it two thirds or half way between the target and the bottom of the screen is fantastic. 

Another consideration is how much head room the center of screen has. If you have the view slightly tilted down, then you already have lost that above view tactically, so then why not move everything up, taking the aim from center-center screen up a bit.

**Comment by tsunamayo** [comment]
Yes I think the best way would be an option inside the entity or the spaceship computer. I am adding this to the to-do list. Thanks
